src/Uno.UI/UI/Xaml/DependencyPropertyDetails.cs

```diff
@@ -13,14 +15,19 @@ namespace Windows.UI.Xaml
 	/// <summary>
 	/// Represents the stack of values used by the Dependency Property Value Precedence system
 	/// </summary>
-	internal class DependencyPropertyDetails : IEnumerable<object>, IEnumerable, IDisposable
+	internal class DependencyPropertyDetails : IEnumerable<object?>, IEnumerable, IDisposable
 	{
 		private DependencyPropertyValuePrecedences _highestPrecedence = DependencyPropertyValuePrecedences.DefaultValue;
-		private BindingExpression _lastBindings;
-		private static readonly ArrayPool<object> _pool = ArrayPool<object>.Create(100, 100);
-		private readonly object[] _stack;
-		private readonly bool _hasWeakStorage;
-		private readonly List<BindingExpression> _bindings = new List<BindingExpression>();
+		private readonly Type _dependencyObjectType;
+		private object? _fastLocalValue;
+		private BindingExpression? _lastBindings;
+		private static readonly ArrayPool<object?> _pool = ArrayPool<object?>.Create(100, 100);
```

Comment:
```suggestion
		private static readonly ArrayPool<object?> _pool = ArrayPool<object?>.Create(_stackLength, 100);
```
I don't know if the `maxArrayLength` is used to create arrays, but we should limit it to the number of precedences we have in the created arrays